File: 2299-MergeNodesInBetweenZeros/2299-MergeNodesInBetweenZeros.java
// Last updated: 04/04/2026, 23:30:25
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeNodes(ListNode head) {
        ListNode temp = head;
        while(temp.next!=null) {
            while(temp.next.val!=0) {
                temp.val += temp.next.val;
                temp.next = temp.next.next;
            }
            if(temp.next.next==null) {
                temp.next = temp.next.next;
                break;
            }
            temp=temp.next;
        }
        return head;
    }
}

File: 2299-MergeNodesInBetweenZeros/2299-MergeNodesInBetweenZeros.cpp
// Last updated: 04/04/2026, 23:30:23
/**
 * Definition for singly-linked list->
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(NULLptr) {}
 *     ListNode(int x) : val(x), next(NULLptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeNodes(ListNode* head) {
        ListNode* temp = head;
        while(temp->next!=NULL) {
            while(temp->next->val!=0) {
                temp->val += temp->next->val;
                temp->next = temp->next->next;
            }
            if(temp->next->next==NULL) {
                temp->next = temp->next->next;
                break;
            }
            temp=temp->next;
        }
        return head;
    }
};